Report, implementation, results

Simpen-1 opened the pull request:
added to the report in the implementation and results part, by simon sandberg and oleksiey michenko

Simpen-1 committed:
Added report writings

report writings about robot arm and line follower, made by simon sandberg and oleksiey michenko

Simpen-1 committed:
added results to report

added results about controllers made for arm motion, by simon sandberg

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
-\subsection{Line follower}
-% GPIO pins does not work. Talk about the dedicated hardware instead.
+\subsection{Robot Hardware}
+The robot designated the task of moving the object on the industry platform is built from the Lego Mindstorms EV3 core set package. The design of the robot is based from the \cite{Robot arm} model from Legos EV3 core set instructions, but is modified with continuous tracks on the base of the robot, longer arm and some reinforcement to make the robot oscillate less when moving. Instead of using th ...
```
"The robot designed **for** the task ..."

tmplt replied:
Also: use `\parencite` instead.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
 
+\subsection{Line Follower}
+% GPIO pins does not work. Talk about the dedicated hardware instead.
+The docking sequence of the robots movement was implemented using a line follower. A QTRX reflectance sensor from Pololu was used together with a black line to guide the robot close to the factory and to stop it with a perpendicular angle ton the factory(pic). During the project, there occured a connection issue between the Raspberry Pi 4?? and the sensor array due to the fact that no libraries t ...
```
* Describe what the docking sequence is.
* Explain what QTRX means.
* "angle ton the factory(pic)" - what?
* Just call it the "Raspberry Pi".
* Libraries were available to talk to the sensor array, but NixOS was the root cause of the issue here. Blame NixOS itself instead.
* "thru" -> "through"
* Add a reference to the breakout board, like you have done to the EV3 hardware.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+The docking sequence of the robots movement was implemented using a line follower. A QTRX reflectance sensor from Pololu was used together with a black line to guide the robot close to the factory and to stop it with a perpendicular angle ton the factory(pic). During the project, there occured a connection issue between the Raspberry Pi 4?? and the sensor array due to the fact that no libraries t ...
+
+\subsubsection{Adafruit FT232H Breakout}
+Adafruit FT232H Breakout is a breakout board from Adafruit which contains the F232H chip from FTDI and has has one port each for a usb c and stemma qt male connectors. The board is also equipped with 16 GPIO pins and also has the ability to speak with many protocols including SPI, ITC, serial UART, JTAg and others. In the project, this board was used to transfer GPIO singals from individual senso ...
```
* "JTAg" -> "JTAG".
* Remove the trailing "The" or finish the sentence.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\subsubsection{Sensor Array}
+The Sensor array used in this project was a QTRX-MD-16A Reflectance array from Pololu. Each sensor on the sensor array is made from a photo transistor and paired with an IR-led which powers on and keeps emitting light while the sensor array module is powered on. The signal from the photo transistor is transmitted as an analog voltage from the sensor array. 
+
+\newpage
```
Remove the `\newpage`. I'll handle the final typesetting before submission.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\subsubsection{Adafruit FT232H Breakout}
+Adafruit FT232H Breakout is a breakout board from Adafruit which contains the F232H chip from FTDI and has has one port each for a usb c and stemma qt male connectors. The board is also equipped with 16 GPIO pins and also has the ability to speak with many protocols including SPI, ITC, serial UART, JTAg and others. In the project, this board was used to transfer GPIO singals from individual senso ...
+\subsubsection{Sensor Array}
+The Sensor array used in this project was a QTRX-MD-16A Reflectance array from Pololu. Each sensor on the sensor array is made from a photo transistor and paired with an IR-led which powers on and keeps emitting light while the sensor array module is powered on. The signal from the photo transistor is transmitted as an analog voltage from the sensor array. 
```
* "led" -> "LED"

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\begin{figure}[H]
+    \centering
+    \includegraphics[width = 5 cm]{Mobile_platform_paint.PNG}
+    \caption{The Mobile Platform seen from above}
```
* Commit this image, but move it to `assets/` instead.
* Finish the caption with a period.
* Use `width=\textwidth` instead.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\newpage
+\subsection {Mobile Platform and Trailer}
+\subsubsection{Mobile Platform}
+A simplified model of the mobile platform can be seen in figure \ref{Mobile_platform_paint}. The power from each motor is transmitted by a beam to three interconnected gears which in turn are inscircled by a rubber band. This configuration  creates two continues tracks powered by one motor each. Due to the fact that each continues track is built in parallel to the platform, the robot is limited t ...
```
* Refer to figures using `Figure~/ref{}` instead.
* "inscircled" -> "encircled".
* "continues" -> "continuous".

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+    \centering
+    \includegraphics[width = 5 cm]{Band_platform_spikes.PNG}
+    \caption{The Mobile Platform seen from the side of one track}
+    \label{Mobile_platform_paint}
```
Ditto, and use a different label; the previous one is overwritten now.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\end{figure}
+
+\subsubsection{Trailer}
+In order to get power to each motor servo and the Raspberry PI, two battery packs with 8 batteries in each were used. Since these battery packs together with the Raspberry Pi and the two BrickPi motorshields were to large to place on the mobile platform, a trailer was built and mounted behind the platform. 
```
* "Raspberry PI" -> "Raspberry Pi";

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\begin{figure}[H]
+    \centering
+    \includegraphics[width = 13 cm]{Arm_model.png}
+    \caption{Arm model side view}
```
Ditto.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\begin{figure}[H]
+    \centering
+    \includegraphics[width = 13 cm]{Arm_overview.png}
+    \caption{Arm model overview}
```
Ditto.

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+In order to get power to each motor servo and the Raspberry PI, two battery packs with 8 batteries in each were used. Since these battery packs together with the Raspberry Pi and the two BrickPi motorshields were to large to place on the mobile platform, a trailer was built and mounted behind the platform. 
+
+\subsection{Robot arm}
+A simplified model of the robot arm can be seen in figure \ref{Arm_model} where the measured distances from \(d_1\) to \(d_7\) and the vertical angle Theta can be seen. in figure \ref{arm_overview} an overview of the robot model can be seen showing how the angle beta was specified.
```
* LaTeX can handle greek letters, so use `\theta` and `\beta` if the same is done in the figures;
* `\(d_1\) to \(d_7\)` -> `$(d_1)$ to $(d_7)$`;
* Fix `in figure \ref{arm_overview}`;

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+    \caption{Arm model overview}
+    \label{arm_overview}
+\end{figure}
+The robot arm can rotate in the xy-plane around the \(z_0\) axis which can be seen in fig. \ref{Arm_model} with a angle Beta using a EV3 large servomotor mounted inside the base of the robot.The upper part of the robot, from wheel1 to the gripper claw can also rotate vertically in the xz-plane using another EV3 large servomotor mounted between wheel0 and the point on the base from where the axis  ...
```
* use `$$` instead of `\( \)`;
* Refer to figures using `Fig~\ref{}` instead;
* use greek letters instead;
* `robot.The upper part of the robot`: sentences are seperated with a space;

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+\subsubsection{Resolution}
+As mentioned before the motors used in the robot arm was two Lego EV3 servo motor large for making the arm move vertically and horizontally and one Lego EV3 servo motor medium inside the gripper to pick up boxes on the miniature industry platform. Both the Large and the medium motors had a resolution of 1 degree, meaning the motors gives 360 readings on one revolution. Also on the robot the cogwh ...
+\subsubsection{Initialization}
+There is no memory in the motors which makes the robot "unaware" of the original orientation of the arm on boot. Therefore an initialization process was made for the robot using two Lego EV3 touch sensors \ref{touch_sensor}. On boot the robot moved the arm with constant angular speed, in the beta direction, until a lego piece, moving with the arm, pressed one of the touch sensors at \(beta = -90^ ...
```
* quotes in LaTeX are done with 
```
``unare''
```
* don't `\newpage`;

tmplt commented on report/sections/implementation.tex:
```diff
@@ -236,9 +236,296 @@ \subsubsection{LCM}
+    beta = sin^{-1}(\frac{y}{(d_5 + d_6)\cdot cos(theta) - d_2\cdot cos(alpha) + d_1})
+    \label{Beta_y_robot}
+\end{equation}
+these two expressions both have a denominator which is equal to zero for an angle \(theta = \pm \pi /2\), two angles that the robot can never reach due to the way the Lego hardware is put together.\\
```
* Don't linebreak (the `\\`)

tmplt replied:
- use `\frac{\pi}{2}` instead;

tmplt reviewed (changes requested):
Here is an initial review. Please see to the review comments. Additionally:

* only capitalize the first letter in the first word of each section title to match the style of the report;
* add to `team-members.tex` which member has written which section(s);
* ignore my `\( \)` -> `$$` comments if the compiled result looks like it should (I haven't seen the use of `'\( \)` before);
* finish all captions with a period;
* commit all your figures into `assets/`, and use `\textwidth` instead of a hardcoded width value;
* don't `\newpage` and don't linebreak (`\\`);

Cascade these changes throughout your additions and I'll do a full review.

Simpen-1 committed:
added to membership

Simpen-1 committed:
added more to team members

Simpen-1 committed:
Added figures to report

Simpen-1 committed:
Added more figures to report

Simpen-1 committed:
added to team members

Simpen-1 committed:
added to conclusion

added to conclusion about robot hardware, controllers and about how the robot could complete it's given task

Simpen-1 committed:
added to improvements

added small bit to improvements about robot arm/hardware

Simpen-1 committed:
Added litterature to ref

Added Feedback Control of Dynamic Systems

Simpen-1 committed:
correction

Simpen-1 committed:
update implementation

Simpen-1 committed:
update results

Simpen-1 committed:
update implementation

Simpen-1 committed:
updated results

added alex michenkos part in results

Simpen-1 committed:
update ref

Simpen-1 committed:
Merge branch 'master' into master

Simpen-1 committed:
update implementation

added "." to fig refs

Simpen-1 committed:
update implementation

corrected equation refering

Simpen-1 reviewed (commented):
done

Simpen-1 committed:
update implementation

equations and figs correctly referred to, changed section "Robot Hardware" to "Robot hardware" removed "\newline"

Simpen-1 committed:
adde files 

added figures

Simpen-1 committed:
update implementation

fixed typos and some other stuff

Simpen-1 committed:
update team members

added Oleksiy Mishchenkos part of team members

Simpen-1 committed:
update conclusions

changed quotations

Simpen-1 committed:
update implementation

corrected referring to equations and removed lcm-source-robot-target.py

Simpen-1 committed:
update results

corrected referring to a figure

Simpen-1 committed:
update implementation

changed sections to only have capital letters in beginning of first word

Simpen-1 committed:
update implementation

defined where to find figures

Simpen-1 committed:
update results

defined where figures are

Simpen-1 committed:
updated results

added "." to the caption missing it and the sentences where it was missing

Simpen-1 committed:
update implementation

removed lcm-source-robot-target

Simpen-1 committed:
update results

corrections of spelling

Simpen-1 committed:
update implementation

correction of typo
